privacyidea/lib/task/base.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#  2018-06-20 Friedrich Weber <friedrich.weber@netknights.it>
+#             Initial skeleton
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+__doc__ = """This is the base class for a task module.
+
+Tasks may be run periodically. Upon execution, a task
+can be given a set of options.
+"""
+
+
+class BaseTask(object):
+    """
+    A BaseTask returns a list of supported options.
+    """
+
+    identifier = "BaseTask"
+    description = "This is the base class of a task with no functionality"
+
+    def __init__(self):
+        pass
+
+    @property
+    def options(self):
+        """
+        A task may be given a dictionary of options. The allowed keys, their
+        descriptions and types are specified here.
+
+        For each option key, the returned dictionary contains one nested dictionary.
+        Each nested dictionary has keys "type", "desc" and "value".
+
+        :return: dict
+        """
+        return {}
+
+    def do(self, options=None):
```

Comment:
The options in the event handlers are not the configuration itself. But also request specific context:

    { "g": g,
      "handler_def": <handler optiones> }

So the naming of parameter ``options`` and the property ``options`` might be a bit misleading, because it is not the same. The property ``options`` returns the information, how the module can be configured. The paramter ``options`` can contain the configuration but it could also contain additional context.

We should keep this in mind. Maybe, when implementing the rest, we realize, that we should change the naming. For now, I think it is ok.

Reply:
Ah, good to know!
Yes, let's see what we'll need. I guess theoretically, we shouldn't need ``g`` because we are operating outside the request context.